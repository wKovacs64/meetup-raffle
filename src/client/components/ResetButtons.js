import React from 'react';
import PropTypes from 'prop-types';

const ResetButtons = ({ onReset, onSubmit }) => (
  <div className="flex flex-wrap justify-around mt4">
    <button
      className="w-100 w5-ns f5 f4-ns b input-reset ba near-black b--near-black bg-white hover-bg-moon-gray pointer ph5 pv3 mb3 mb0-l shadow-5"
      type="button"
      onClick={onReset}
    >
      Reset
    </button>
    <button
      className="w-100 w5-ns f5 f4-ns b input-reset ba near-black b--near-black bg-white hover-bg-moon-gray pointer ph5 pv3 mb3 mb0-l shadow-5"
      type="button"
      onClick={onSubmit}
    >
      Draw Again
    </button>
  </div>
);

ResetButtons.propTypes = {
  onReset: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default ResetButtons;
