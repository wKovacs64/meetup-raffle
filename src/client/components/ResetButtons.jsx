import PropTypes from 'prop-types';

function ResetButtons({ onReset, onRetry }) {
  return (
    <div className="my-8 flex flex-shrink-0 flex-wrap justify-around">
      <button
        className="mb-4 w-full border border-solid border-current bg-white py-4 px-16 font-bold shadow-lg hover:bg-gray-300 focus:bg-gray-300 sm:mb-0 sm:w-64 sm:text-xl"
        type="button"
        onClick={onReset}
      >
        Start Over
      </button>
      <button
        className="w-full border border-solid border-current bg-white py-4 px-16 font-bold shadow-lg hover:bg-gray-300 focus:bg-gray-300 sm:w-64 sm:text-xl"
        type="button"
        onClick={onRetry}
      >
        Draw Again
      </button>
    </div>
  );
}

ResetButtons.propTypes = {
  onReset: PropTypes.func.isRequired,
  onRetry: PropTypes.func.isRequired,
};

ResetButtons.displayName = 'ResetButtons';

export default ResetButtons;
