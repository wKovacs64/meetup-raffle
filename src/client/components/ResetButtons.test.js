import React from 'react';
import { render, Simulate } from 'react-testing-library';
import ResetButtons from './ResetButtons';

describe('ResetButtons', () => {
  const onReset = jest.fn();
  const onSubmit = jest.fn();

  const { container, getByText } = render(
    <ResetButtons onReset={onReset} onSubmit={onSubmit} />,
  );

  it('renders', () => {
    expect(container).toMatchSnapshot();
  });

  it('calls onReset appropriately', () => {
    expect(onReset).toHaveBeenCalledTimes(0);
    Simulate.click(getByText('Reset'));
    expect(onReset).toHaveBeenCalledTimes(1);
  });

  it('calls onSubmit appropriately', () => {
    expect(onSubmit).toHaveBeenCalledTimes(0);
    Simulate.click(getByText('Draw Again'));
    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
