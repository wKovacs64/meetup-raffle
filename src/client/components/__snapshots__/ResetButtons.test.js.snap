// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetButtons renders 1`] = `
<div>
  <div
    class="flex flex-wrap justify-around mt4"
  >
    <button
      class="w-100 w5-ns f5 f4-ns b input-reset ba near-black b--near-black bg-white hover-bg-moon-gray pointer ph5 pv3 mb3 mb0-l shadow-5"
      type="button"
    >
      Reset
    </button>
    <button
      class="w-100 w5-ns f5 f4-ns b input-reset ba near-black b--near-black bg-white hover-bg-moon-gray pointer ph5 pv3 mb3 mb0-l shadow-5"
      type="button"
    >
      Draw Again
    </button>
  </div>
</div>
`;
